modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/ClayTable.tsx

```diff
@@ -0,0 +1,661 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton, {ClayButtonWithIcon} from '@clayui/button';
+import {Body, Cell, Head, Row, Table} from '@clayui/core';
+import {ClayCheckbox, ClayRadio} from '@clayui/form';
+import ClayLoadingIndicator from '@clayui/loading-indicator';
+import {useIsMounted} from '@liferay/frontend-js-react-web';
+import {FDSTableCellHTMLElementBuilderArgs} from '@liferay/js-api/data-set';
+import classNames from 'classnames';
+import {ClientExtension} from 'frontend-js-components-web';
+import {throttle} from 'frontend-js-web';
+import React, {useContext, useEffect, useMemo, useRef, useState} from 'react';
+
+import {IItemsActions} from '../..';
+import FrontendDataSetContext, {
+	IFrontendDataSetContext,
+	TRenderer,
+} from '../../FrontendDataSetContext';
+import Actions from '../../actions/Actions';
+import {getInternalCellRenderer} from '../../cell_renderers/getInternalCellRenderer';
+import {
+	ILocalizedItemDetails,
+	getLocalizedValue,
+} from '../../utils/getLocalizedValue';
+import {getInputRendererById} from '../../utils/renderer';
+import ViewsContext, {
+	IViewsContext,
+	TViewsContextDispatch,
+} from '../ViewsContext';
+
+// @ts-ignore
+
+import {VIEWS_ACTION_TYPES} from '../viewsReducer';
+import {InlineEditInputRenderer} from './TableCell';
+import TableContext from './dnd_table/TableContext';
+
+type Field = {
+	contentRenderer: string;
+	contentRendererClientExtension: boolean;
+	fieldName: any;
+	label: string;
+	localizeLabel: boolean;
+	mapData: Function;
+	sortable: boolean;
+};
+
+type Props = {
+	fields: Array<Field>;
+	inlineAddingSettings?: {
+		apiURL?: string;
+		defaultBodyContent?: Record<string, any>;
+	};
+	itemInlineChanges?: Array<any>;
+	items: Array<any>;
+	itemsActions: Array<IItemsActions>;
+	nestedItemsReferenceKey?: string;
+	selectItems: Function;
+	selectable?: boolean;
+	selectedItemsKey: string;
+	selectedItemsValue: any;
+	selectionType?: string;
```

Comment:
Would it make more sense to selectionType be a boolean for multiselect, since we are always just checking if it's equal or not equal to "multiple"?

Reply:
> Would it make more sense to selectionType be a boolean for multiselect, since we are always just checking if it's equal or not equal to "multiple"?

Yes, I believe so, a lot of the code actually I just tried to keep the same and change as little as possible so this comes from the APIs. I'm not sure if there are other options but I don't think so because a lot of the code is not well typed.

Reply:
Okay that makes sense. Yep it's best to not change the api unless necessary.